### PR TITLE
[MU4] Added telemetry configuration

### PR DESCRIPTION
**Pull request by igorkorsukov**
Added telemetry configuration

### CODE REVIEW

File: framework/telemetry/telemetrysetup.cpp [hunk 1]
```diff
@@ -35,6 +36,8 @@
 
 using namespace mu::telemetry;
 
+static std::shared_ptr<TelemetryConfiguration> s_conf = std::make_shared<TelemetryConfiguration>();
```
Comment (RomanPudashkin) [inline framework/telemetry/telemetrysetup.cpp:39]: Better rename to s_configuration

File: framework/telemetry/internal/telemetryconfiguration.h [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+#ifndef MU_TELEMETRY_TELEMETRYCONFIGURATION_H
+#define MU_TELEMETRY_TELEMETRYCONFIGURATION_H
+
+#include "../itelemetryconfiguration.h"
+
+namespace mu::telemetry {
+class TelemetryConfiguration : public ITelemetryConfiguration
+{
+public:
+    TelemetryConfiguration() = default;
```
Comment (RomanPudashkin) [inline framework/telemetry/internal/telemetryconfiguration.h:28]: This constructor will be generated automatically

**Review by RomanPudashkin — APPROVED** [review]

File: framework/telemetry/telemetrysetup.cpp [hunk 3]
```diff
@@ -52,6 +55,7 @@ void TelemetrySetup::registerResources()
 
 void TelemetrySetup::registerExports()
 {
+    mu::framework::ioc()->registerExport<ITelemetryConfiguration>("telemetry", s_conf);
```
Comment (Eism) [inline framework/telemetry/telemetrysetup.cpp:58]: better to use moduleName() method

File: framework/telemetry/itelemetryconfiguration.h [hunk 4]
```diff
@@ -0,0 +1,38 @@
+//=============================================================================
+//  MuseScore
+//  Music Composition & Notation
+//
+//  Copyright (C) 2019 MuseScore BVBA and others
```
Comment (Eism) [inline framework/telemetry/itelemetryconfiguration.h:5]: please, update the license :)

File: framework/telemetry/telemetrysetup.cpp (hunk 1 above)
Comment (igorkorsukov) [reply framework/telemetry/telemetrysetup.cpp]: renamed

File: framework/telemetry/internal/telemetryconfiguration.h (hunk 2 above)
Comment (igorkorsukov) [reply framework/telemetry/internal/telemetryconfiguration.h:28]: I prefer explicit over implicit.

File: framework/telemetry/telemetrysetup.cpp (hunk 3 above)
Comment (igorkorsukov) [reply framework/telemetry/telemetrysetup.cpp]: fixed

File: framework/telemetry/itelemetryconfiguration.h (hunk 4 above)
Comment (igorkorsukov) [reply framework/telemetry/itelemetryconfiguration.h]: fixed

**Review by Eism — APPROVED** [review]

**Pull request merged**